api/logout_test.go

```diff
@@ -0,0 +1,56 @@
+package api
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/netlify/gotrue/conf"
+	"github.com/netlify/gotrue/models"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type LogoutTestSuite struct {
+	suite.Suite
+	API    *API
+	Config *conf.GlobalConfiguration
+	token  string
+}
+
+func TestLogout(t *testing.T) {
+	api, config, err := setupAPIForTest()
+	require.NoError(t, err)
+
+	ts := &LogoutTestSuite{
+		API:    api,
+		Config: config,
+	}
+	defer api.db.Close()
+
+	suite.Run(t, ts)
+}
+
+func (ts *LogoutTestSuite) SetupTest() {
+	models.TruncateAll(ts.API.db)
+
+	u, err := models.NewUser("", "test@example.com", "password", ts.Config.JWT.Aud, nil)
+	require.NoError(ts.T(), err, "Error creating test user model")
+	require.NoError(ts.T(), ts.API.db.Create(u), "Error saving new test user")
+
+	// generate access token to use for logout
+	t, err := generateAccessToken(u, nil, time.Second*time.Duration(ts.Config.JWT.Exp), ts.Config.JWT.Secret)
+	require.NoError(ts.T(), err)
+	ts.token = t
+}
+
+func (ts *LogoutTestSuite) TestLogoutSuccess() {
+	req := httptest.NewRequest(http.MethodPost, "http://localhost/logout", nil)
+	req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", ts.token))
+	w := httptest.NewRecorder()
+
+	ts.API.handler.ServeHTTP(w, req)
+	require.Equal(ts.T(), http.StatusNoContent, w.Code)
```

Comment:
Can you check that the cookies are deleted in the response? I think that's the important bit.